mlx/primitives.h

```diff
@@ -2158,4 +2158,36 @@ class Cholesky : public UnaryPrimitive {
   bool upper_;
 };
 
+class Eigvalsh : public UnaryPrimitive {
```

Comment:
Sorry if @barronalex already made this suggestion: but I think it makes sense to merge the `Eighvalsh` and `Eigh` into a single primitive. And have the ops use the same primitive but just return only the eigenvalues in the case of `eighvalsh`. 

It looks like the work is done anyway.. and the underlying implementations are basically identical.